### PR TITLE
Cleans up check_streak() code

### PR DESCRIPTION
**Pull request by ghost**
check_streak() is now a proc innate to all martial arts
removes constant repeating of by making it run after the check_streak() code which escapes with return 0 on a failed check
```
streak = ""
return 1
```

Tested with chef CQC and krav maga since they behave differently on check_streak(), both worked as intended

### CODE REVIEW

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Can you make all of them into switches or at least consistent with each other.

**Comment by McBawbaggings** [comment]
no, because streaks aren't the same length so
```
#define ATTACK_COMBO "hhd"
switch(streak)
     if(ATTACK_COMBO)
```
where streak is "dhhd" then it won't work

**Comment by ExcessiveUseOfCobblestone** [comment]
what about elseifs?

**Comment by McBawbaggings** [comment]
I don't follow, you want me to change the switch statement for the action button martial arts into if/if else/else statements like the combo martial arts for consistency purposes or am I misunderstanding?

**Comment by ExcessiveUseOfCobblestone** [comment]
I would like them to all be consistent in the format, yes.

**Comment by McBawbaggings** [comment]
okay, will do

File: code/datums/martial/cqc.dm [hunk 1]
```diff
@@ -20,27 +20,21 @@
...
+	else if(findtext(streak,PRESSURE_COMBO))
 		Pressure(A,D)
-		return 1
-	if(findtext(streak,CONSECUTIVE_COMBO))
-		streak = ""
+	else if(findtext(streak,CONSECUTIVE_COMBO))
 		Consecutive(A,D)
-	return 0
+	else
+		return 0
```
Comment (ShizCalev) [inline code/datums/martial/cqc.dm]: FALSE

File: code/datums/martial/cqc.dm [hunk 2]
```diff
@@ -20,27 +20,21 @@
...
+	streak = ""
+	return 1
```
Comment (ShizCalev) [inline code/datums/martial/cqc.dm]: TRUE

File: code/datums/martial/krav_maga.dm [hunk 3]
```diff
@@ -69,21 +69,18 @@
...
-			streak = ""
 			leg_sweep(A,D)
-			return 1
 		if("quick_choke")//is actually lung punch
-			streak = ""
 			quick_choke(A,D)
-			return 1
-	return 0
+		else
+			return 0
```
Comment (ShizCalev) [inline code/datums/martial/krav_maga.dm]: FALSE

File: code/datums/martial/krav_maga.dm [hunk 4]
```diff
@@ -69,21 +69,18 @@
...
+	streak = ""
+	return 1
```
Comment (ShizCalev) [inline code/datums/martial/krav_maga.dm]: TRUE

File: code/datums/martial/plasma_fist.dm [hunk 5]
```diff
@@ -7,20 +7,17 @@
...
-	if(findtext(streak,PLASMA_COMBO))
-		streak = ""
+	else if(findtext(streak,PLASMA_COMBO))
 		Plasma(A,D)
-		return 1
-	return 0
+	else
+		return 0
+	streak = ""
+	return 1
```
Comment (ShizCalev) [inline code/datums/martial/plasma_fist.dm]: TRUE

File: code/datums/martial/sleeping_carp.dm [hunk 6]
```diff
@@ -11,28 +11,21 @@
...
 		headKick(A,D)
-		return 1
-	if(findtext(streak,ELBOW_DROP_COMBO))
-		streak = ""
+	else if(findtext(streak,ELBOW_DROP_COMBO))
 		elbowDrop(A,D)
-		return 1
-	return 0
+	else
+		return 0
```
Comment (ShizCalev) [inline code/datums/martial/sleeping_carp.dm]: FALSE

File: code/datums/martial/sleeping_carp.dm [hunk 7]
```diff
@@ -11,28 +11,21 @@
...
+	streak = ""
+	return 1
```
Comment (ShizCalev) [inline code/datums/martial/sleeping_carp.dm]: TRUE

File: code/datums/martial/wrestling.dm [hunk 8]
```diff
@@ -16,29 +16,22 @@
...
-			streak = ""
 			throw_wrassle(A,D)
-			return 1
 		if("slam")
-			streak = ""
 			slam(A,D)
-			return 1
-	return 0
+		else
+			return 0
```
Comment (ShizCalev) [inline code/datums/martial/wrestling.dm]: FALSE

File: code/datums/martial/wrestling.dm [hunk 9]
```diff
@@ -16,29 +16,22 @@
...
+	streak = ""
+	return 1
```
Comment (ShizCalev) [inline code/datums/martial/wrestling.dm]: TRUE

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
@McBawbaggings

**Comment by optimumtact** [comment]
stale

**Pull request closed without merging**